pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,454 @@
+	cmd.Flags().String("cert", "", "Path to a certificate file.")
+	cmd.Flags().String("key", "", "Path to a key file.")
+	cmd.Flags().String("ca-cert", "", "Path to a CA certificate file.")
+	cmd.Flags().String("dest-ca-cert", "", "Path to a CA certificate file, used for securing the connection from the router to the destination.")
+	cmd.MarkFlagRequired("dest-ca-cert")
+
+	return cmd
+}
+
+// CreateReencryptRoute implements the behavior to run the create reencrypt route command.
+func CreateReencryptRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	serviceName, err := resolveServiceName(f, kcmdutil.GetFlagString(cmd, "expose"))
+	if err != nil {
+		return err
+	}
+	routeName, err := resolveRouteName(args)
+	if err != nil {
+		return err
+	}
+	route, err := unsecuredRoute(kc, ns, routeName, serviceName, kcmdutil.GetFlagString(cmd, "port"))
+	if err != nil {
+		return err
+	}
+
+	route.Spec.Host = kcmdutil.GetFlagString(cmd, "hostname")
+	route.Spec.Path = kcmdutil.GetFlagString(cmd, "path")
+
+	route.Spec.TLS = new(api.TLSConfig)
+	route.Spec.TLS.Termination = api.TLSTerminationEdge
+
+	cert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Certificate = string(cert)
+	key, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "key"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Key = string(key)
+	caCert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "ca-cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.CACertificate = string(caCert)
+	destCACert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "dest-ca-cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.DestinationCACertificate = string(destCACert)
+
+	route, err = oc.Routes(ns).Create(route)
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+	resourceMapper := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: resource.ClientMapperFunc(f.Factory.RESTClient),
+	}
+	info, err := resourceMapper.InfoForObject(route)
+	if err != nil {
+		return err
+	}
+	shortOutput := kcmdutil.GetFlagString(cmd, "output") == "name"
+	kcmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, "created")
+	return nil
+}
+
+// unsecuredRoute will return a route with enough info so that it can direct traffic to
+// the service provided by --expose. Callers of this helper are responsible for providing
+// tls configuration, path, and the hostname of the route.
+func unsecuredRoute(kc *kclient.Client, namespace, routeName, serviceName, portString string) (*api.Route, error) {
+	if len(routeName) == 0 {
+		routeName = serviceName
+	}
+
+	svc, err := kc.Services(namespace).Get(serviceName)
+	if err != nil {
+		return &api.Route{
```

Comment:
> It was quickly identified as a bug when multiport services came along.

@kargakis I'm torn.  I could see the argument that a simple service doesn't require a port on the route, but I'm not sure that I want to encourage that.  I guess I see these commands as intent-based, so I wouldn't mind changing what they produce if this turns out to be hard to use.